### PR TITLE
NetworkWidget: Add Blocking column

### PR DESCRIPTION
**Pull request by sepalani**
This PR adds a "Blocking" column to the Network widget telling if the socket is blocking.

Ready to be reviewed & merged.

### CODE REVIEW

File: Source/Core/Core/IOS/Network/Socket.cpp [hunk 1]
```diff
@@ -652,6 +652,11 @@ s32 WiiSockMan::AddSocket(s32 fd, bool is_rw)
   return wii_fd;
 }
 
+bool WiiSockMan::IsSocketBlocking(s32 wii_fd) const
+{
+  return WiiSockets.count(wii_fd) != 0 && !WiiSockets.at(wii_fd).nonBlock;
```
Comment (iwubcode) [inline Source/Core/Core/IOS/Network/Socket.cpp:657]: Would a `find` be more performant?  I think having both of these calls does two lookups.

**Comment by sepalani** [comment]
@iwubcode 
You're right. I fixed it.

File: Source/Core/DolphinQt/Debugger/NetworkWidget.cpp [hunk 2]
```diff
@@ -90,6 +90,15 @@ QTableWidgetItem* GetSocketState(s32 host_fd)
   return new QTableWidgetItem(QTableWidget::tr("Unbound"));
 }
 
+QTableWidgetItem* GetSocketBlocking(s32 wii_fd)
+{
+  const auto& socket_manager = IOS::HLE::WiiSockMan::GetInstance();
+  if (socket_manager.GetHostSocket(wii_fd) < 0)
+    return new QTableWidgetItem();
```
Comment (Tilka) [inline Source/Core/DolphinQt/Debugger/NetworkWidget.cpp:97]: Not saying it should be removed, but why are you checking for a host socket?

Comment (sepalani) [reply Source/Core/DolphinQt/Debugger/NetworkWidget.cpp:97]: The function is called for each possible socket fd. Some are used by the Wii and some aren't. The one which aren't will return `-EBADF`, that's why I'm checking the GetHostSocket return value. The same check is done in many get functions above: `if (host_fd < 0)`.

**Pull request merged**